src/codeu/chat/server/Server.java

```diff
@@ -123,6 +124,20 @@ public void onMessage(InputStream in, OutputStream out) throws IOException {
       }
     });
 
+    // New Interest - A client wants to add a new interest to the back end.
+    this.commands.put(NetworkCode.NEW_INTERESTS_REQUEST,  new Command() {
+      @Override
+      public void onMessage(InputStream in, OutputStream out) throws IOException {
+
+        final Uuid userid = (Uuid.SERIALIZER).read(in);
+        final Uuid interest = (Uuid.SERIALIZER).read(in);
+        final Interests interests = controller.newInterest(userid, interest);
+
+        Serializers.INTEGER.write(out, NetworkCode.NEW_INTERESTS_RESPONSE);
+        Serializers.nullable(Interests.SERIALIZER).write(out, interest);
```

Comment:
This should be `interests` because you're returning the full Interests object for the user.